packages/mongodb-dialects/spring-criteria/src/main/kotlin/com/mongodb/jbplugin/dialects/springcriteria/AggregationStagesParser.kt

```diff
@@ -0,0 +1,69 @@
+package com.mongodb.jbplugin.dialects.springcriteria
+
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiMethod
+import com.intellij.psi.PsiMethodCallExpression
+import com.mongodb.jbplugin.dialects.javadriver.glossary.fuzzyResolveMethod
+import com.mongodb.jbplugin.dialects.javadriver.glossary.resolveToMethodCallExpression
+import com.mongodb.jbplugin.dialects.springcriteria.aggregationparsers.MatchStageParser
+import com.mongodb.jbplugin.mql.Node
+
+/**
+ * Parser for parsing supported patterns of writing an aggregation pipeline.
+ * Supported patterns for writing aggregation calls are the following:
+ * 1. MongoTemplate.aggregate()
+ * 2. MongoTemplate.aggregateStream()
+ *
+ * The AggregationParser concerns itself only with parsing the aggregation related semantics and
+ * leave the rest as a responsibility for the composing unit.
+ */
+class AggregationStagesParser(private val matchStageParser: MatchStageParser) {
+    private fun isStageCall(stageCallMethod: PsiMethod): Boolean {
+        return MatchStageParser.isMatchStageCall(stageCallMethod)
+    }
+
+    private fun parseAggregationStages(
+        newAggregationCall: PsiMethodCallExpression
+    ): List<Node<PsiElement>> {
+        val newAggregationCallArguments = newAggregationCall.argumentList.expressions
+        val resolvedStageCalls = newAggregationCallArguments.mapNotNull { stageCallExpression ->
+            stageCallExpression.resolveToMethodCallExpression { _, stageCallMethod ->
+                isStageCall(stageCallMethod)
+            }
+        }
+        return resolvedStageCalls.map { stageCall ->
+            val stageCallMethod = stageCall.fuzzyResolveMethod() ?: return@map Node(
+                source = stageCall,
+                components = emptyList()
+            )
+
+            if (MatchStageParser.isMatchStageCall(stageCallMethod)) {
```

Comment:
Can we change this method to be an instance method like `parse`? From what I understand with this design, each stage parser will be an object, so if we put this method in an interface and rename it 'canParse' or something like that, we can use the same interface for all parsers and get consistency and change this if to a for loop over an array of instances.